server/src/main/java/org/apache/druid/segment/realtime/appenderator/AppenderatorPlumber.java

```diff
@@ -153,6 +153,7 @@ public IncrementalIndexAddResult add(InputRow row, Supplier<Committer> committer
   {
     final SegmentIdWithShardSpec identifier = getSegmentIdentifier(row.getTimestampFromEpoch());
     if (identifier == null) {
+      metrics.reportMessageMaxTimestamp(row.getTimestampFromEpoch());
```

Comment:
@kaijianding Did you mean this to help with the Kafka indexing service mechanism? If so, this isn't the right file; it's only used by `AppenderatorPlumberSchool`, but Kafka indexing doesn't use that. I think AppenderatorImpl would be the right place to add it.